core/controllers/topics_and_skills_dashboard.py

```diff
@@ -166,6 +166,36 @@ def get(self):
         self.render_json(self.values)
 
 
+class UnassignSkillDataHandler(base.BaseHandler):
+    """Provides data for Unassign skill modal."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_topics_and_skills_dashboard
+    def get(self, skill_id):
+        """Handles GET requests."""
+        skill_ids_assigned_to_some_topic = (
+            topic_services.get_all_skill_ids_assigned_to_some_topic())
+
+        assigned_topics = {}
+        topics = topic_fetchers.get_all_topics()
```

Comment:
Ditto about fetching all topics. Since line 178 fetches all topics to do the check.
Also, you don't really need that function anymore here, right?

Reply:
RIght, fixed!